bigchaindb/core.py

```diff
@@ -449,15 +454,22 @@ def has_previous_vote(self, block):
             True if this block already has a valid vote from this node, False otherwise. If
             there is already a vote, but the vote is invalid, raises an ImproperVoteError
         """
-        if block['votes']:
-            for vote in block['votes']:
-                if vote['node_pubkey'] == self.me:
-                    if util.verify_vote_signature(block, vote):
-                        return True
-                    else:
-                        raise ImproperVoteError('Block {block_id} already has an incorrectly signed vote '
-                                                'from public key {me}').format(block_id=block['id'], me=self.me)
-        return False
+        votes = r.table('votes').get_all([block['id'], self.me], index='block_and_voter').run(self.conn)
+
+        votes = list(votes)
+
+        if len(votes) > 1:
+            raise MultipleVotesError('Block {block_id} has {n_votes} votes from public key {me}'
+                                     .format(block_id=block['id'], n_votes=str(len(votes)), me=self.me))
+        has_previous_vote = False
+        for vote in votes:
```

Comment:
This is a self-check on the node to see if it has more than one vote on a particular block.  In normal operation, no this should not happen.

If you still think I should remove it, no problem.